jscomp/ml/typecore.ml

```diff
@@ -662,6 +664,9 @@ let rec collect_missing_arguments env type1 type2 = match type1 with
       | Some res -> Some ((label, argtype) :: res)
       | None -> None
     end
+  | t when Ast_uncurried_utils.typeIsUncurriedFun t -> 
+    let typ = Ast_uncurried.typeExtractUncurriedFun t in 
+    collect_missing_arguments env typ type2
```

Comment:
This was the second issue where it should have given the suggestion:

```
  This value might need to be wrapped in a function that takes an extra
  parameter of type unit
                                                                                                                                            
  Here's the original error message
  This has type: option<'a>
  But it's expected to have type: (. unit) => option<int>
                                                                                                                                            
```
it only outputted

```
  This has type: option<'a>
  But it's expected to have type: (. unit) => option<int>
```

since it would not match on the cases which treat it as  a function. We first need to extract the type of the uncurried function.